Utilities/Hash.cpp

```diff
@@ -0,0 +1,89 @@
+#include "stdafx.h"
+#include "Hash.h"
+#include <sstream>
+#include <iomanip>
+
```

Comment:
I'd like to avoid Hash.cpp, it can be incorporated into fs::get_data_dir function.